src/ScopeProvider.tsx

```diff
@@ -1,108 +1,191 @@
 import { createContext, useContext, useState } from 'react';
 import type { ReactNode } from 'react';
-import { Provider, useStore } from 'jotai/react';
-import type { Atom, WritableAtom } from 'jotai/vanilla';
+import { Provider } from 'jotai/react';
+import { Atom, WritableAtom, getDefaultStore } from 'jotai/vanilla';
 
 type AnyAtom = Atom<unknown>;
 type AnyWritableAtom = WritableAtom<unknown, unknown[], unknown>;
-type GetScopedAtom = <T extends AnyAtom>(anAtom: T) => T;
+type InterceptedAtomCopy<T extends AnyAtom> = T & { original?: T };
+type GetInterceptedAtomCopy = <T extends AnyAtom>(anAtom: T) => T;
+type GetStoreKey = <T extends AnyAtom>(anAtom: T) => T;
 
+const isSelfAtom = (atom: AnyAtom, a: AnyAtom) =>
+  atom.is ? atom.is(a) : a === atom;
+const getOriginalOfCopy = (a: InterceptedAtomCopy<AnyAtom>) => a.original ?? a;
 const isEqualSet = (a: Set<unknown>, b: Set<unknown>) =>
   a === b || (a.size === b.size && Array.from(a).every((v) => b.has(v)));
-
-export const ScopeContext = createContext<GetScopedAtom>((a) => a);
-
+type Store = ReturnType<typeof getDefaultStore>;
+export const ScopeContext = createContext<readonly [GetStoreKey, Store]>([
+  (a) => a,
+  getDefaultStore(),
```

Comment:
Just a nit, but `getDefaultStore` is designed with lazy initialization in mind, but this forces to initialize it at the module load.

Reply:
https://github.com/jotaijs/jotai-scope/pull/20/commits/de7a60484deab9a8bd788d6acf792c81bb6d82b2